beanie/odm/fields.py

```diff
@@ -1,5 +1,6 @@
+from __future__ import annotations
```

Comment:
I remember there was some issue a while back with using this import and with lets say Pydantic.
This basically enables all new typing features and whatnot.

I'm fine with this addition if it doesn't break any downstream consumers (existing codebases).

On another note, does this mean that we can now start modernizing the Beanie codebase using modern typing features, such as `list[str] | None` instead of `Optional[List[str]]` and so on (by adding this import to every file)?

Reply:
it should be fine for python 3.8+